[FX] Pass module's qualname to is_leaf_module

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44966 [FX] Pass module's qualname to is_leaf_module**

Heard some demand for this, seems reasonable

Differential Revision: [D23790360](https://our.internmc.facebook.com/intern/diff/D23790360)

Commit:
[FX] Pass module's qualname to is_leaf_module

[ghstack-poisoned]

dzhulgakov reviewed (approved):
Awesome! Please update references in fbcode too (I think there are some)

Commit:
Update on "[FX] Pass module's qualname to is_leaf_module"


Heard some demand for this, seems reasonable

Differential Revision: [D23790360](https://our.internmc.facebook.com/intern/diff/D23790360)

[ghstack-poisoned]
